activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -20,6 +20,31 @@
 
 module ActiveSupport
   module Cache
+    module ConnectionPoolLike
+      def with
+        yield self
+      end
+    end
+
+    ::Redis.include(ConnectionPoolLike)
```

Comment:
Can't we do the same we did with Redis::Distrubuted here?

Reply:
Looks like we can. I'm updating now

Reply:
Sorry, I was incorrect. Unfortunately, we can't use subclass here because user can provide already initialized instance of Redis here https://github.com/rails/rails/blob/fd04c8cbb6627ef65ea728b5026b21bd24471c18/activesupport/lib/active_support/cache/redis_cache_store.rb#L103-L121

I updated my PR accordingly to merged #31424. If all other looks fine, will squash/rebase as needed.